File: languages/go/entity/encryption/aes_encrypter.go
package encryption

import (
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
)

type AesEncrypter struct {
	Alg  proto.EncryptionAlg
	Args EncrypterArgs
}

func NewAesEncrypter(args EncrypterArgs) AesEncrypter {
	return AesEncrypter{
		Alg:  proto.EncryptionAlg_A256GCM,
		Args: args,
	}
}

func (s AesEncrypter) ToProto() *proto.Encrypter {
	var localKey *proto.LocalKey
	if s.Args.LocalKey != nil {
		localKey = s.Args.LocalKey.ToProto()
	} else if s.Args.Key != "" {
		localKey = key.LocalKey{
			Key: s.Args.Key,
		}.ToProto()
	}

	var managedKey *proto.ManagedKey
	if s.Args.ManagedKey != nil {
		managedKey = s.Args.ManagedKey.ToProto()
	}

	return &proto.Encrypter{
		Alg:        s.Alg,
		LocalKey:   localKey,
		ManagedKey: managedKey,
	}
}

File: languages/java/src/main/java/com/bloock/sdk/entity/encryption/AesEncrypter.java
package com.bloock.sdk.entity.encryption;

import com.bloock.sdk.bridge.proto.EncryptionEntities;
import com.bloock.sdk.bridge.proto.EncryptionEntities.EncryptionAlg;
import com.bloock.sdk.entity.key.KeyType;
import com.bloock.sdk.entity.key.LocalKey;
import com.bloock.sdk.entity.key.ManagedKey;

public class AesEncrypter implements Encrypter {
  EncryptionAlg alg;
  EncrypterArgs args;

  public AesEncrypter(String password) {
    this.alg = EncryptionAlg.A256GCM;
    this.args = new EncrypterArgs(new LocalKey(password, null, KeyType.Aes256));
  }

  public AesEncrypter(LocalKey localKey) {
    this.alg = EncryptionAlg.A256GCM;
    this.args = new EncrypterArgs(localKey);
  }

  public AesEncrypter(ManagedKey managedKey) {
    this.alg = EncryptionAlg.A256GCM;
    this.args = new EncrypterArgs(managedKey);
  }

  @Override
  public EncryptionEntities.Encrypter toProto() {
    EncryptionEntities.Encrypter.Builder builder =
        EncryptionEntities.Encrypter.newBuilder().setAlg(this.alg);

    if (this.args.localKey != null) {
      builder.setLocalKey(this.args.localKey.toProto());
    }

    if (this.args.managedKey != null) {
      builder.setManagedKey(this.args.managedKey.toProto());
    }

    return builder.build();
  }
}
